hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -299,20 +299,19 @@ private void getTransaction(DeletedBlocksTransaction tx,
       DatanodeDetails datanodeDetails = replica.getDatanodeDetails();
       if (!dnList.contains(datanodeDetails)) {
         DatanodeDetails dnDetail = replica.getDatanodeDetails();
-        LOG.debug("Skip Container = {}, because DN= {} is not in dnList.",
+        LOG.debug("Skip Container = {}, because DN = {} is not in dnList.",
             containerId, dnDetail.getUuid());
+        metrics.incrSkippedTransaction();
         return;
       }
     }
 
     for (ContainerReplica replica : replicas) {
       DatanodeDetails details = replica.getDatanodeDetails();
-      if (!dnList.contains(details)) {
-        continue;
-      }
       if (!transactionStatusManager.isDuplication(
           details, updatedTxn.getTxID(), commandStatus)) {
         transactions.addTransactionToDN(details.getUuid(), updatedTxn);
+        metrics.incrProcessedTransaction();
```

Comment:
I think we can move the metrics ProcessedTransaction to the begin of the method `getTransaction`. Then base on the `SkippedTransaction/ProcessedTransaction` we can know that the rate of the skipped Transaction

Reply:
I will continue to improve this part of the code.